sdk/inc/azure/core/az_http.h

```diff
@@ -221,13 +221,34 @@ AZ_NODISCARD az_result az_http_response_get_status_line(
     az_http_response* ref_response,
     az_http_response_status_line* out_status_line);
 
+/**
+ * @brief Returns the #az_http_status_code information from an HTTP response.
+ * @details Invokes #az_http_response_get_status_line(), so it advances the reading position
+ * accordingly.
+ *
+ * @remark Use this function when HTTP Reason Phrase is not needed, and when it is not important for
+ * the invoking code to distinguish between #AZ_HTTP_STATUS_CODE_NONE and any possible error when
+ * parsing an HTTP response.
+ *
+ * @param[in,out] ref_response The #az_http_response with an HTTP response.
+ *
+ * @return An HTTP status code.
+ */
+AZ_INLINE az_http_status_code az_http_response_get_status_code(az_http_response* ref_response)
+{
+  az_http_response_status_line status_line = { 0 };
+  return az_result_failed(az_http_response_get_status_line(ref_response, &status_line))
+      ? AZ_HTTP_STATUS_CODE_NONE
+      : status_line.status_code;
+}
```

Comment:
That's fair if you want to give a hint to the compiler.

That said, shouldn't it be marked as `AZ_NODISCARD`? If the user is calling this method, they probably should do something meaningful with the return value, right?

Reply:
We introduced AZ_NODISCARDs specifically for az_results - so that users handle errors, and you will soon run into a problem if you don't analyze them. For other, results, there is no harm in ignoring the result.

Reply:
Removing AZ_NODISCARD is non-breaking, but adding it back later is breaking. I think we should add it here given the scenario. It doesn't really make much sense for a user to call this API and ignore the return value. We should help them, especially since we want to encourage them to check the status code for success before continuing processing the response.

@JeffreyRichter what do you think?